recorder/src/main/cpp/perf_ctx.cc

```diff
@@ -103,10 +103,15 @@ void PerfCtx::ThreadTracker::exit(PerfCtx::TracePt pt) throw (IncorrectEnterExit
         }
     }
 
+    if (actual_stack.size() == 0) {
```

Comment:
One can't push it into that corner with auto-closable style usage (try-with-resources construct).

That test repros it by using it the geek way.